js/views/chatAppVw.js

```diff
@@ -0,0 +1,267 @@
+              }
+            }
+            chat.set('avatarURL', model.get('serverUrl') + "get_image?hash=" + chat.get('image_hash') + "&guid=" + chat.get('guid'));
+            self.renderChat(chat);
+          });
+          $('#chatHeads').html(self.listWrapper);
+        }
+      }
+    });
+
+    this.listenTo(window.obEventBus, "socketMessageRecived", function(response){
+      this.handleSocketMessage(response);
+    });
+
+    this.listenTo(window.obEventBus, "openChat", function(guid, key) {
+      this.openChat(guid, key);
+    });
+
+  },
+
+  render: function(){
+    "use strict";
+    var self = this;
+
+    loadTemplate('./js/templates/chatApp.html', function(loadedTemplate) {
+      self.$el.html(loadedTemplate());
+    });
+
+    return this;
+  },
+
+  renderChat: function(model){
+    var chat = new chatView({
+      model: model
+    });
+    this.subViews.push(chat);
+    this.listWrapper.prepend(chat.el);
+  },
+
+  renderChatMessage: function(model){
+    var chatMessage = new chatMessageView({
+      model: model
+    });
+    this.subViewsChat.push(chatMessage);
+    this.listWrapperChat.prepend(chatMessage.el);
+  },
+
+  renderNoneFound: function(){
+    // Decide what to do here
+  },
+
+  refreshConversations: function() {
+
+  },
+
+  openChat: function(guid, key) {
+    var self = this;
+    this.newChat();
+    $('#inputConversationRecipient').val(guid);
+    $('#inputConversationKey').val(key);
+    $('#inputConversationMessage').focus();
+
+    // Load conversation from DB
+    this.chatMessages = new chatMessageCollection();
+    this.chatMessages.url = this.options.model.get('serverUrl') + "get_chat_messages?guid=" + guid;
+    var model = this.options.model;
+
+    this.listWrapperChat = $('<div class="border0 custCol-border-secondary flexRow"></div>');
+
+    this.chatMessages.fetch({
+      success: function(chatMessages, response) {
+        if(chatMessages.models.length < 1) {
+          console.log('none found');
+        } else {
+          __.each(chatMessages.models, function (chatMessage) {
+            "use strict";
+            if(chatMessage.image_hash === undefined) {
+              var hash = window.localStorage.getItem("avatar_" + chatMessage.get('guid'));
+              if(hash !== "") {
+                chatMessage.set('image_hash', hash);
+              }
+            }
+            if(chatMessage.get('outgoing')) {
+              chatMessage.set('avatarURL', model.get('serverUrl') + "get_image?hash=" + model.get('avatar_hash'));
+            } else {
+              chatMessage.set('avatarURL', model.get('serverUrl') + "get_image?hash=" + chatMessage.get('image_hash') + "&guid=" + chatMessage.get('guid'));
+            }
+            self.renderChatMessage(chatMessage);
+          });
+
+          $('#chatConversation .chatConversationContent').html(self.listWrapperChat).animate({ scrollTop: $('#chatConversation .chatConversationContent').prop("scrollHeight")}, 100);
```

Comment:
Change this to 
$('#chatConversation .chatConversationContent').html(self.listWrapperChat).promise().done(function() {
            "use strict";
            $(this).animate({ scrollTop: numberOfChatMessages*100}, 100);
          });

The issue here has two parts. Animate needs to happen in a promise so it isn't triggered before the DOM is changed. 

The other part is scrollTop is being measured wrong. When you scrollTop to the height of the parent (145px), you're only scrolling down by 145px, and the child element may be much taller than that. 

But we can't just get the height of the child element, because even with a promise it's not actually on the DOM yet and has no height. So instead I am suggesting counting the number of messages, and scrolling by 100px per message. That should always be enough, unless we enable shift+enter and someone has messages with lots of soft returns. 

If you scroll by more than the height of the child element, it's fine, it'll just scroll to the bottom. We could also just scroll to an arbitrarily large number, like 9999999999999999999 (which works as long as it's in the promise);